### DISCUSSION TITLE
Generating Rule for custom Object detection

### DISCUSSION THREAD

**Discussion by Abiganesh** [AxisCommunications/discussions#184] [ACAP] [answered]
I have built a custom object detection application using ACAP native sdk examples, and I want to trigger an event whenever an object is detected. To do that, I have to generate a rule, but on the rules page, my custom object detection condition is not listed. Please kindly guide me to add a condition to the rules page for my custom object detection model.

**Comment by onie5** [comment]
Hi Abiganesh,

Have you published your events?

Take a look at the events example on how to expose your events for the rule engine.

Cheers

**Comment by daniel-falk** [comment, marked as answer]
@Abiganesh  When declaring your events you need to use the right topology for it to be visible in the cameras event UI. You can chose to use the Axis namespace or the ONVIF namespace. Here is an example that works:
![Slide41](https://user-images.githubusercontent.com/23218783/227710102-eceee5d8-ceb4-4472-a0bc-1f077699922a.jpg)
![GetImage](https://user-images.githubusercontent.com/23218783/227710190-7d597cac-f24b-40c6-914a-757402615aae.png)
